docs/contributor/adr/0007-rate-limit.md

```diff
@@ -0,0 +1,30 @@
+# Rate Limit
+
+## Status
+
+Draft
+
+## Context
+
+It has been decided to indroduce rate limit functionality into Kyma platform to allow users to set rate limit functionality on the service mesh layer, therefore allowing to consume intended service mesh functionality which is abstracting away networking concerns outside applications inside the mesh.
+Since Istio is an underlying service mesh responsible for the workloads networking across Kyma clusters, therefore it is the Istio which allows to configure such functionality.
+At this moment it is not possible to introduce global rate limit provided by Istio service mesh because of the technical or legal limitations, related to the deployment of key-value stores in the cluster.
+Major difference between local and global rate limit is that local rate limiting is applied directly at the sidecar proxy level, controlling the rate of requests on a per-instance basis. Global rate limiting is applied accross multiple instances of a service or across, multiple services, providing a coorinated rate limit within the mesh.
+In case of introduction the global rate limit functionality in the future, new ADR will be created.
+
+## Decision
```

Comment:
Do we want to return the rate limit headers per default?
https://github.com/kyma-project/istio/blob/rate-limit-poc/docs/contributor/draft/rate-limiting-poc/01-rate-limiting.md#rate-limit-response-headers

Reply:
The only risk I can see in having those headers is that it gives a possibility to fine tune DoS attacks to the rate limit values to make detection harder. Second thing is potential insight into how many requests are getting to the system. I'd turn them off by default as the most secure option, and give users possibility to turn it on with one simple field.

Reply:
The field would be a boolean `enableHeaders` on the level of `spec`?

Reply:
I think we can do enableHeaders. I'll add it to the decision